pkg/scheduler/state/state.go

```diff
@@ -371,8 +372,8 @@ func withReserved(key types.NamespacedName, podName string, committed int32, res
 
 func isPodUnschedulable(pod *v1.Pod) bool {
 	annotVal, ok := pod.ObjectMeta.Annotations[scheduler.PodAnnotationKey]
-	unschedulable, val := strconv.ParseBool(annotVal)
-	return ok && val == nil && unschedulable
+	unschedulable, err := strconv.ParseBool(annotVal)
+	return (ok && err == nil && unschedulable) || pod.Spec.NodeName == ""
```

Comment:
Could we please add a short comment on this line just to remind us later that `pod.Spec.NodeName == ""` happens in a case when pods are in pending state?

Reply:
+1